listenbrainz_spark/hdfs/__init__.py

```diff
@@ -0,0 +1,89 @@
+import os
+import sys
+import time
+import shutil
+import tempfile
+import subprocess
+
+import listenbrainz_spark
+from listenbrainz_spark import utils, config, hdfs_connection
+from listenbrainz_spark.exceptions import SparkSessionNotInitializedException
+
+from flask import current_app
+
+class ListenbrainzHDFSUploader:
+
+    def __init__(self):
+        hdfs_connection.init_hdfs(config.HDFS_HTTP_URI)
+        try:
+            listenbrainz_spark.init_spark_session('uploader')
+        except SparkSessionNotInitializedException as err:
+            current_app.logger.error(str(err), exc_info=True)
+            sys.exit(-1)
+
+    def _is_json_file(self, filename):
+        """ Check if passed filename is a JSON file
+
+        Args:
+            filename (str): the name of the file
+
+        Returns:
+            bool: True if JSON file, False otherwise
+        """
+        return filename.endswith('.json')
+
+    def get_pxz_output(self, archive, threads=8):
+        """ Spawn a new pxz process to decompress tar.
+
+            Args:
+                archive: Tar to decompress.
+                threads: Maximal number of threads to run simultaneously.
+
+            Returns:
+                pxz: Return pipe to pxz command.
+        """
+        pxz_command = ['pxz', '--decompress', '--stdout', archive, '-T{}'.format(threads)]
+        pxz = subprocess.Popen(pxz_command, stdout=subprocess.PIPE)
+        return pxz
+
+    def upload_archive(self, tar, dest_path, schema, callback=None, force=False):
+        """ Upload data dump to HDFS.
+
+            Args:
+                tar: Uncompressed tar object.
+                dest_path (str): HDFS path to upload data dump.
+                schema: Schema of parquet to be uploaded.
+                callback: Function to process JSON files.
+                force: If True deletes dir at dest_path
+        """
+        if callback is None:
+            current_app.logger.critical('Callback to process JSON missing. Aboritng...')
```

Comment:
Python has a NotImplementedException for this -- we should use that here. This is grave enough of an error to warrant stopping everything.